src/client/pydaos/pydaos_shim.c

```diff
@@ -0,0 +1,238 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+ * The Government's rights to use, modify, reproduce, release, perform, display,
+ * or disclose this software are subject to the terms of the Apache License as
+ * provided in Contract No. B609815.
+ * Any reproduction of computer software, computer software documentation, or
+ * portions thereof marked with this legend must also reproduce the markings.
+ */
+
+#ifdef __USE_PYTHON3__
+#include <python3.6m/Python.h>
+
+/* Those are gone from python3, replaced with new functions */
+#define PyInt_FromLong		PyLong_FromLong
+#define PyInt_AsLong		PyLong_AsLong
+#define PyString_FromString	PyUnicode_FromString
+#define PyString_AsString	PyBytes_AsString
+#else
+#include <Python.h>
+#endif
+
+#include "daos_types.h"
+#include "daos.h"
+#include "daos_obj_class.h"
+#include <daos/object.h>
+
+#define PY_SHIM_MAGIC_NUMBER 0x7A89
+
+static int
+__is_magic_valid(int input)
+{
+	if (input != PY_SHIM_MAGIC_NUMBER) {
+		D_ERROR("MAGIC number doesnt match, expected %d got %d\n",
+			PY_SHIM_MAGIC_NUMBER, input);
+		return 0;
+	}
+
+	return 1;
+}
+
+/** Macro that parses out magic value and verifies it */
+#define RETURN_NULL_IF_BAD_MAGIC(args)					\
+do {									\
+	int magic;							\
+	if (!PyArg_ParseTuple(args, "i", &magic)) {			\
+		D_ERROR("Bad arguments passed to %s\n", __func__);	\
+		return NULL;						\
+	}								\
+									\
+	if (!__is_magic_valid(magic)) {					\
+		return NULL;						\
+	}								\
+} while (0)
+
+
+/** Parse arguments and magic number out*/
+#define RETURN_NULL_IF_FAILED_TO_PARSE(args, format, x...)		\
+do {									\
+	int magic;							\
+	if (!PyArg_ParseTuple(args, "i"format, &magic, x)) {		\
+		D_ERROR("Bad arguments passed to %s\n", __func__);	\
+		return NULL;						\
+	}								\
+									\
+	if (!__is_magic_valid(magic)) {					\
+		return NULL;						\
+	}								\
+ } while (0)
+
+/**
+ * Implementations of shim functions
+ */
+
+static PyObject *
+__shim_handle__test(PyObject *self, PyObject *args)
+{
+	long ret;
+
+	RETURN_NULL_IF_BAD_MAGIC(args);
+
+	/* Call C function */
+	ret = 1;
+
+	return PyInt_FromLong(ret);
+}
+
+/**
+ * Exports of enumerations/macros
+ */
+
+/** Object Class Definition */
+static void
+oc_define(PyObject *module)
+{
+#define DEFINE_OC(pref, suf) \
+	PyModule_AddIntConstant(module, "OC_" #pref #suf, OC_##pref##suf)
+
+#define DEFINE_OC_PROT(prot)	\
+do {				\
+	DEFINE_OC(prot, TINY);	\
+	DEFINE_OC(prot, SMALL);	\
+	DEFINE_OC(prot, LARGE);	\
+	DEFINE_OC(prot, MAX);	\
+} while (0)
+
+	DEFINE_OC_PROT();		/** OC_TINY, OC_SMALL, ... */
+	DEFINE_OC_PROT(RP_);		/** OC_RP_TINY, OC_RP_SMALL, ... */
+	DEFINE_OC_PROT(RP_SF_);		/** OC_RP_SF_TINY, ... */
+	DEFINE_OC(RP_XSF,);		/** OC_RP_XSF */
```

Comment:
(style)  space required after that ',' (ctx:VxB)

Reply:
Moved it out of this DEFINE_OC_PROT() bundle.